src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private static final int DAY = 0;
+    private static final int MONTH = 1;
+    private static final int YEAR = 2;
+    private static final int WORK_DAY = 0;
+    private static final int EMP_NAME = 1;
+    private static final int WORK_HOURS = 2;
+    private static final int PER_HOUR = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate from = parseToLocalDate(dateFrom);
+        LocalDate to = parseToLocalDate(dateTo);
+        int[] salary = new int[names.length];
+        for (int i = 0; i < data.length; i++) {
+            String[] splittedData = data[i].split(" ");
+            for (int j = 0; j < names.length; j++) {
+                LocalDate dateOfWork = parseToLocalDate(splittedData[WORK_DAY]);
+                if (dateOfWork.compareTo(from) >= 0 && dateOfWork.compareTo(to) <= 0
+                        && splittedData[EMP_NAME].equals(names[j])) {
+                    salary[j] += Integer.parseInt(splittedData[WORK_HOURS])
+                            * Integer.parseInt(splittedData[PER_HOUR]);
+                }
+            }
+        }
```

Comment:
There is also an option to get rid of the inner loop and improve performance. This is optional, but it would be cool if you do)

Reply:
You can use HashMap for example ;)